ktp/src/main/kotlin/toothpick/ktp/KTP.kt

```diff
@@ -25,32 +25,23 @@ import toothpick.ktp.delegate.DelegateNotifier
 /**
  * Main Toothpick API entry point for Kotlin.
  */
-class KTP : Toothpick() {
+object KTP {
 
-    companion object TP {
-        val delegateNotifier = DelegateNotifier()
+    val delegateNotifier = DelegateNotifier()
 
-        init {
-            injector = object : InjectorImpl() {
-                override fun <T : Any> inject(obj: T, scope: Scope) =
-                    if (delegateNotifier.hasDelegates(obj)) {
-                        delegateNotifier.notifyDelegates(obj, scope)
-                    } else {
-                        super.inject(obj, scope)
-                    }
-            }
-        }
+    init {
+        InjectorReplacer.replace()
```

Comment:
cannot do it cause of visiblitiy. Otherwise we would need to make the field protected (instead of package-private) and extend Toothpick (which means, having to go back to companion and `KTP.TP....`)